app/vue/src/client/preview/types.ts

```diff
@@ -7,8 +7,10 @@ export interface ShowErrorArgs {
   description: string;
 }
 
-// TODO: some vue expert needs to look at this
-export type StoryFnVueReturnType = string | Component;
+/** Any options that can be passed into Vue.extend */
+export type VueOptions = Exclude<Parameters<VueConstructor['extend']>[0], undefined>;
```

Comment:
I could not find a union of all options interfaces in the Vue typings, pulling all possible arguments to `extend` gets them all quite easily. The types are very long otherwise. Let me know if you know a better type.